components/admin_console/team_channel_settings/group/group_users/group_users_role.tsx

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {UserProfile} from 'mattermost-redux/types/users';
+import {TeamMembership} from 'mattermost-redux/types/teams';
+import {ChannelMembership} from 'mattermost-redux/types/channels';
+
+import {Group} from 'mattermost-redux/types/groups';
+
+import * as Utils from 'utils/utils';
+
+type ProfileWithGroups = Partial<UserProfile & {
+    groups: Partial<Group>[];
+}>;
+
+interface GroupUsersRoleProps {
+    user: ProfileWithGroups;
+    membership: TeamMembership | ChannelMembership;
+    scope: 'team' | 'channel';
+}
+
+type Role = 'system_admin' | 'team_admin' | 'team_user' | 'channel_admin' | 'channel_user' | 'guest';
+export default class GroupUsersRole extends React.PureComponent<GroupUsersRoleProps, {}> {
```

Comment:
Same question here since it's relatively simple. Plus, if you use a functional component, you can use the `useIntl` hook instead of `Utils.localizeMessage` since the latter isn't particularly React-friendly.

It'd look something like
```typescript
import {useIntl} from 'react-intl';

export default function GroupUsersRole(props: GroupUsersRoleProps) {
    const intl = useIntl();

    const role = getCurrentRole(props);

    let localizedRole;
    switch (role) {
        localizedRole = intl.formatMessage({id: 'admin.user_grid.system_admin', defaultMessage: 'System Admin'});
    }

    // ...
}
```

Reply:
Also, the empty object for the `State` type isn't necessary. You can just omit that type parameter for class components.